### PR TITLE
DEV-2133 | Make ReactAppView not fail when no social_meta

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?

Fixes a bug in ReactAppView whereby the view would fail when a RevenueProgram didn't have an associated SocialMeta instance (which lies in one-to-one relation).

Adds a test for this case.

#### Why are we doing this? How does it help us?

So donation pages won't be broken for RPs that don't have social meta.

#### How should this be manually tested? Please include detailed step-by-step instructions.

In NRH admin, find the RP for a live donation page. Delete that RP's social meta instance (which can be done from the detail view for the RP).  

Now visit the donation page. It should load. If you inspect the head of the HTML file in developer tools, none of the social meta related values should appear (the values are pictured below, but this is what you shouldn't see on this page):

Screen Shot 2022-08-02 at 2 39 22 PM

#### Have automated unit tests been added? If not, why?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-2133](https://news-revenue-hub.atlassian.net/jira/software/c/projects/DEV/boards/3?modal=detail&selectedIssue=DEV-2133)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No

### CODE REVIEW

**Review by njharman — APPROVED** [review]
Looks good. Tests pass. Could not check coverage because ./revengine/ isn't part of test coverage (fix that in other ticket/follow on to DEV-1808)

The except branch is covered so lgtm!

**Comment by njharman** [comment]
Needs to have main merged into it

File: pyproject.toml [hunk 1]
```diff
@@ -71,6 +71,7 @@ whitenoise = "^6.2.0"
 django-reversion-compare = "^0.15.0"
 django-migration-linter = "^4.1.0"
 django-test-migrations = "^1.2.0"
+pytest-mock = "^3.8.2"
```
Comment (x110dc) [inline pyproject.toml:74]: This should be a dev dep, no?

**Review by x110dc — CHANGES_REQUESTED** [review]
One minor request.

File: pyproject.toml (hunk 1 above)
Comment (x110dc) [reply pyproject.toml]: This is here but it's also in dev deps. I removed it from here.

**Review by x110dc — APPROVED** [review]

**Pull request merged**